lib/libvmod_blob/hex.c

```diff
@@ -99,22 +99,35 @@ hex_encode(const enum encoding enc, const enum case_e kase,
 		*p++ = alphabet[in[i] & 0x0f];
 	}
 
+	/* XXX: the length arithmetics here are a bit complicated for such
+	 * a simple problem. It looks like the estimated encoded length
+	 * makes room for a null character, which makes sense for string
+	 * encoding. But null characters are set by the caller as documented.
+	 *
```

Comment:
This XXX-comment is strangely placed, since it doesn't seem to be about the code that it's close to.

Don't see what's complicated or wrong. Two hex chars to a byte. hex_encode_l is 2*nbytes + 1, with the one for the terminating null. hex_decode_l is nbytes/2, rounding up (that's the odd-numbered hex string case).